rename-overwrite/index.js

```diff
@@ -49,6 +49,18 @@ module.exports = async function renameOverwrite (oldPath, newPath, retry = 0) {
         await fs.promises.mkdir(path.dirname(newPath), { recursive: true })
         await renameOverwrite(oldPath, newPath, retry)
         break
+      // Crossing filesystem boundaries so rename is not available
+      case 'EXDEV':
+        try {
+          await rimraf(newPath)
+        } catch (rimrafErr) {
+          if (rimrafErr.code !== 'ENOENT') {
+            throw rimrafErr
+          }
+        }
+        await fs.promises.copyFile(oldPath, newPath)
```

Comment:
I guess this won't work with directories

Reply:
nice catches